dali/test/python/decoder/test_imgcodec.py

```diff
@@ -0,0 +1,420 @@
+# Copyright (c) 2019-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
1. Are those tests brand new or a modified version of existing tests for the existing decoder?
2. Do you test the expermiental peek_shape?
3. If one replaced all the fn.[op] with fn.experimental.[op] in the existing tests, would that work?

Reply:
1, 3. Those tests are the copy of the existing tests (that's why I left 2019 in the copyright), with the following changes:
- I replaced  fn.[op] with fn.experimental.[op], as you describe
- I commented out the parts that don't work yet to make the tests pass
I'd demonstrate that with a diff, but I can't find an option to diff selected files on github :/

2. Kind of... It's used in two tests in the file, so it _somehow_ works, but it would probably deserve a test of its own

Reply:
Added peek_shape tests 9a4625d4d2bfb2b70c46831f57c1ea6a7e6175c2